### PR TITLE
Initial files and Longhorn Engine rock.

### PR DESCRIPTION
**Pull request by aznashwan**

### CODE REVIEW

File: tests/integration/test_longhorn_in_helm_chart.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        "csi-node-driver-registrar": "v2.11.1",
+        "csi-resizer": "v1.11.1",
+        "csi-snapshotter": "v7.0.2",
+        "livenessprobe": "v2.13.1",
+        "openshift-origin-oauth-proxy": "4.15",
+    }
+}
+
+# This mapping indicates which fields of the upstream Longhorn Helm chart
+# contain the 'image' fields which should be overriden with the ROCK
```
Comment (claudiubelu) [inline tests/integration/test_longhorn_in_helm_chart.py]: typo, causing ``lint`` to fail: ``overriden ==> overridden``.

File: tests/integration/test_longhorn_in_helm_chart.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+    # HACK(aznashwan): all the Longhorn deployment resources in the chart
+    # take a while to be properly set up, so we provide a generous wait:
+    retry_kwargs = {"retry_times": 30, "retry_delay_s": 10}
+
+    daemonsets = [
+        "longhorn-csi-plugin",
+        "longhorn-manager",
+    ]
+    for stateful_set in daemonsets:
```
Comment (claudiubelu) [inline tests/integration/test_longhorn_in_helm_chart.py]: ``stateful_set`` => ``daemonset``, since we're iterating over daemonsets.

File: v1.7.0/longhorn-engine/rockcraft.yaml [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+  # The setup steps for the build container are contained within this Dockerfile:
+  # https://github.com/longhorn/longhorn-engine/blob/v1.7.0/Dockerfile.dapper
+  setup-build-env:
+    plugin: nil
+
+    build-packages:
+      # https://github.com/longhorn/longhorn-engine/blob/v1.7.0/Dockerfile.dapper#L30-L36
+      - cmake
+      - curl
+      - git
```
Comment (claudiubelu) [inline v1.7.0/longhorn-engine/rockcraft.yaml:49]: we should already have ``git`` installed, it couldn't clone otherwise.

Comment (claudiubelu) [reply v1.7.0/longhorn-engine/rockcraft.yaml:49]: Nvm, git doesn't seem to be installed by default.

File: v1.7.0/longhorn-engine/rockcraft.yaml [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+      - autoconf
+      - libtool
+      - libunwind-dev
+
+    override-build: |
+      set -eux -o pipefail
+
+      # NOTE(aznashwan): pull and install libqcow2 for building the engine:
+      # https://github.com/longhorn/longhorn-engine/blob/v1.7.0/Dockerfile.dapper#L48-L53
+      curl -sSfL https://s3-us-west-1.amazonaws.com/rancher-longhorn/libqcow-alpha-20181117.tar.gz | tar xvzf - -C /usr/src
```
Comment (berkayoz) [inline v1.7.0/longhorn-engine/rockcraft.yaml:93]: I wonder if we could use libqcow from the ubuntu archive or build from source, I'm fine with having this as a start but we should look into updating this later on.

File: v1.7.0/longhorn-engine/rockcraft.yaml [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+      - libaio-dev
+      - sg3-utils
+      # NOTE(aznashwan): original image install 'iputils' SLE package
+      # which contains the utilities from all the below packages:
+      - iputils-arping
+      - iputils-ping
+      - iputils-tracepath
+      - iputils-clockdiff
+      - iproute2
+      # - qemu-tools
```
Comment (berkayoz) [inline v1.7.0/longhorn-engine/rockcraft.yaml]: Any reason for commenting this out?

**Review by berkayoz — COMMENTED** [review]
Great work, left some comments

File: v1.7.0/longhorn-engine/rockcraft.yaml (hunk 4 above)
Comment (aznashwan) [reply v1.7.0/longhorn-engine/rockcraft.yaml:93]: We absolutely could, though I intentionally wanted to download and use the Rancher-vendored one for compatibility reasons in case they have added/changed any functionality on their fork.

The only thing that could prevent us using the Ubuntu-packaged lib is compatibility issues (the version in Noble is `20201213` vs Rancher's `20181117`), though I did NOT try using the .deb to confirm one way or the other...

File: v1.7.0/longhorn-engine/rockcraft.yaml (hunk 5 above)
Comment (aznashwan) [reply v1.7.0/longhorn-engine/rockcraft.yaml]: Great catch! I had left that as commented to check whether the whole `qemu-tools` package was actually needed or not since i had a hunch it didn't.

TL;DR: there is a `longhornctl` subcommand which calls the `qemu-img` binary, so `qemu-img` is the only thing that's needed, not the whole contents of the `qemu-tools` SLES package.

Context:
* [this Engine commit](https://github.com/longhorn/longhorn-engine/commit/54accb6a0) added the Longhorn subcommand which needs `qemu-img`
  * that commit also [first adds the `qemu-utils` Ubuntu 16.04](https://github.com/longhorn/longhorn-engine/commit/54accb6a0#diff-c55d1ee74f32cd4b9eeeceaec981a180f52e761acb158a5b0b0e088c7725c9caL4-R4) package dep (back then the images were based on Trusty)
* [this Engine commit](https://github.com/longhorn/longhorn-engine/commit/2b055aa80#diff-c55d1ee74f32cd4b9eeeceaec981a180f52e761acb158a5b0b0e088c7725c9caL7-R13) switched the image building from Ubuntu 16.04 to SLES BCI and thus translated the `qemu-utils` Ubuntu Deb to `qemu-tools` SLES RPM
* the [`qemu-tools` SLE package](https://opensuse.pkgs.org/15.6/opensuse-oss-x86_64/qemu-tools-8.2.2-150600.1.8.x86_64.rpm.html) contains a bunch of random binaries/scripts but depends on a separate SLE package named `qemu-img`

As such, I'll update it to simply install `qemu-utils` on Ubuntu.

File: v1.7.0/longhorn-engine/rockcraft.yaml [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+      - gcc
+      - python3
+      - python3-pip
+      - python3-dev
+      - libkmod-dev
+      - libnl-3-dev
+      - linux-headers-6.8.0-41
+      - pkg-config
+      - psmisc
+      # - qemu-tools
```
Comment (berkayoz) [inline v1.7.0/longhorn-engine/rockcraft.yaml]: Do we need to replace this with `qemu-utils` too ?

Comment (aznashwan) [reply v1.7.0/longhorn-engine/rockcraft.yaml]: Great catch, no, I've just removed it!

The reason it was there to begin with is that the Longhorn folks use the Dapper-built Docker container image for both building and integration testing, so they basically install every runtime dependency in it, [including `qemu-img`](https://github.com/longhorn/longhorn-engine/blob/master/Dockerfile.dapper#L31).

File: v1.7.0/longhorn-engine/rockcraft.yaml (hunk 4 above)
Comment (berkayoz) [reply v1.7.0/longhorn-engine/rockcraft.yaml:93]: We can keep this as is for the start and keep it in mind, maybe adding a TODO is helpful.

**Review by berkayoz — APPROVED** [review]
Great work, LGTM
